tests/test_visualization.py

```diff
@@ -322,8 +322,36 @@ def test_svg_processor_with_heralds_phys(tmp_path, save_figs):
     _save_or_check(p, tmp_path, sys._getframe().f_code.co_name, save_figs, recursive=True)
 
 
-# For now, this is broken, but once the heralds change is merged, this will
-# generate a correct image.
+def test_svg_processor_with_heralds_margin_overflow_left_phys(tmp_path, save_figs):
+    c = pcvl.Circuit(3) // BS() // (1, BS())
+    pc = pcvl.Processor('SLOS', c)
+    pc.add_herald(0,0)
+    _save_or_check(pc, tmp_path, sys._getframe().f_code.co_name, save_figs, recursive=True)
+
+
+def test_svg_processor_with_heralds_margin_overflow_right_phys(tmp_path, save_figs):
+    c = pcvl.Circuit(4) // (1, BS()) //  BS()
+    pc = pcvl.Processor('SLOS', c)
+    pc.add_herald(0,0)
+    pc.add_herald(2,1)
+    _save_or_check(pc, tmp_path, sys._getframe().f_code.co_name, save_figs, recursive=True)
+
+
+def test_svg_processor_with_heralds_margin_overflow_left_right_phys(tmp_path, save_figs):
+    c = pcvl.Circuit(2) // BS()
+    pc = pcvl.Processor('SLOS', c)
+    pc.add_herald(0,0)
+    _save_or_check(pc, tmp_path, sys._getframe().f_code.co_name, save_figs, recursive=True)
+
+
+def test_svg_processor_with_heralds_perm_following_phys(tmp_path, save_figs):
+    c = pcvl.Circuit(4) // (1, PERM([1, 0])) // (1, BS()) // (0, PERM([1, 0])) // BS() // (1, PERM([1, 0]))
+    pc = pcvl.Processor('SLOS', c)
+    pc.add_herald(0,0)
+    pc.add_herald(2,1)
+    _save_or_check(pc, tmp_path, sys._getframe().f_code.co_name, save_figs, recursive=True)
+
+
 def test_svg_processor_with_heralds_and_barriers_phys(tmp_path, save_figs):
```

Comment:
This test fails in this branch now that heralds have moved (I think you predicted it in the `Barrier` branch :) )

Also, even when fixing the SVG reference, there seems to be a glitch where the zones left and right of the gray barrier are white (don't take the background color into account)
![image](https://github.com/Quandela/Perceval/assets/49883965/c2e02b10-51b6-448c-bef2-6dd4d8dc38ee)

Could you please fix this?

Reply:
That's strange, the reason why I added `set_background_color` to canvas was precisely to this: made an element aware of the background on which it is drawn. I'll look into it!